### ISSUE TITLE
Problem parsing json for responses with no body (i.e. 504)

### ISSUE THREAD

**Issue by gcweeks** [plaid/plaid-ruby#137]
The `run` method in `lib/plaid/connector.rb` goes something like this:

```ruby
@response = http.request(@request)

# All responses are expected to have a JSON body, so we always parse,
# not looking at the status code.
@body = MultiJson.load(@response.body)
```

My issue is that occasionally I get returned a 504, presumably because of excess server load (I'm on the sandbox if that's relevant). Whatever the reason, this returns a `@response` with an empty string for the body, which throws a `MultiJson` error because an empty string cannot be parsed. This causes the entire method call to crash rather than returning a `Plaid::ServerError` or something that I'm expecting to handle.

I was thinking about opening a pull request that just checks to see `if @response.body.blank?`, but I saw later on in the `raise_error` method, you explicitly rely on the body to report the error, so I'm not sure how you guys wanted to handle the case in which the response has no body...

**Comment by rbuckheit** [comment]
:fish: looks good to me, thanks @be9!

**Comment by be9** [comment]
@gcweeks the problem is fixed in the gem version 2.1.0 which has just been cut!
